publish/pub.go

```diff
@@ -49,17 +50,12 @@ type Publisher struct {
 }
 
 type PendingReq struct {
-	Transformables []transform.Transformable
-	Tcontext       *transform.Context
+	Transformables []Transformable
 	Trace          bool
 }
 
-// PublisherConfig is a struct holding configuration information for the publisher,
-// such as shutdown timeout, default pipeline name and beat info.
-type PublisherConfig struct {
-	Info            beat.Info
-	ShutdownTimeout time.Duration
-	Pipeline        string
```

Comment:
Removing the configuration and instead making the `publish` package aware of the `config.config`  is not a necessary change for this refactoring, and I don't think it is an improvement so please revert.